oauthenticator/azuread.py

```diff
@@ -64,6 +65,16 @@ def get_tenant(self):
             app_log.info('ID4: {0}'.format(tenant_id))
             return tenant_id
 
+    def get_username_claim(self):
+        """
+        The claim to map to the jupyter username, such as `upn` or `unique_name`
+        See https://docs.microsoft.com/en-gb/azure/active-directory/develop/id-tokens
+        """
+        if hasattr(self, 'username_claim') and self.username_claim:
+            app_log.info('ID5: {0}'.format(self.username_claim))
+            return self.username_claim
+        return 'oid'
```

Comment:
The default used to be `'name'`, unless there is a good reason I wouldn't change this. If we do change it we need to think about if or not this will break current users (and then how to deal with that)

Reply:
Makes sense, changed.